velox/functions/lib/Re2Functions.h

```diff
@@ -39,11 +39,23 @@ enum class PatternKind {
   kPrefix,
   /// Fixed pattern preceded by one or more '%', such as '%foo', '%%%hello'.
   kSuffix,
+  /// Patterns matching '%{c0}%', such as '%foo%%', '%%%hello%'.
+  kFixedMiddle,
   /// Patterns which do not fit any of the above types, such as 'hello_world',
   /// '_presto%'.
   kGeneric,
 };
 
+struct PatternMetadata {
+  PatternKind patternKind;
+  // Contains the length of the fixed pattern for patterns of kind kFixed,
+  // kPrefix, and kSuffix. Contains the count of wildcard character '_' for
+  // patterns of kind kExactlyN and kAtLeastN. Contains 0 otherwise.
+  vector_size_t length;
+  // Contains the fixed pattern in patterns of kind kOptimizedPattern.
```

Comment:
kOptimizedPattern is replaced by kFixedMiddle now.

Reply:
Updated, thanks for pointing it out.